Helm: add relevant pod annotations for certificate injecting via Vault Agent (#4660)
grafana/mimir: docs/sources/helm-charts/mimir-distributed/configure/configure-hashicorp-vault-agent.md

  enabled: true
  roleName: "test-role"
  clientCertPath: "client/cert/path"
  clientKeyPath: "client/key/path"
  serverCertPath: "server/cert/path"
  serverKeyPath: "server/key/path"
  caCertPath: "ca/cert/path"
```

Example deployment yaml:

```yaml
spec:
  template:
    metadata:
      annotations:
        vault.hashicorp.com/agent-inject: "true"
        vault.hashicorp.com/role: "test-role"
        vault.hashicorp.com/agent-inject-secret-client.crt: "client/cert/path"
        vault.hashicorp.com/agent-inject-secret-client.key: "client/key/path"
        vault.hashicorp.com/agent-inject-secret-server.crt: "server/cert/path"
        vault.hashicorp.com/agent-inject-secret-server.key: "server/key/path"
        vault.hashicorp.com/agent-inject-secret-root.crt: "ca/cert/path"
```

`vault.hashicorp.com/agent-inject-secret-<FILENAME>: '<PATH>'` tells the Vault Agent where to find the secret, and the name of the file to write the secret to. For example: `vault.hashicorp.com/agent-inject-secret-client.crt: 'client/cert/path'` will look for the secret at the path `client/cert/path` within Vault, and mount this secret to the pod as `client.crt` in the `/vault/secrets/` directory.

> For more information about Vault and Vault Agent, see: https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar

Reviewer: In most (if not all) documentation, link is in form of [title](url). Maybe can consider following this instead having raw url for consistency.

> To configure TLS in Mimir, see: https://grafana.com/docs/mimir/latest/operators-guide/secure/securing-communications-with-tls/

Reviewer: For internal URL usually it is in format [title]({{< relref "../relative-path-to-the-doc"}})

---

> title: "Configure Grafana Mimir to receive Client/Server Certificates and Keys via Hashicorp Vault Agent"

Reviewer 2: ```suggestion
title: "Configure Grafana Mimir to receive client and server certificates and keys via Hashicorp Vault Agent"
```

menuTitle: "Vault Agent"

> description: "Learn how to configure Grafana Mimir to allow Vault Agent to inject secrets into pods"

Reviewer 2: ```suggestion
description: "Learn how to configure Grafana Mimir to allow Vault Agent to inject secrets into Pods."
```

---

# Configure Grafana Mimir to receive Client/Server Certificates and Keys via Hashicorp Vault Agent

> When enabled, the Pod annotations for TLS configurable pods will be updated. Upon deployment, Vault Agent will intercept the Deployment, fetch the relevant secrets from Vault and mount them to the Pod. Note: Vault and Vault Agent are required to be running already.

Reviewer 2: ```suggestion
When you enable this feature, Helm updates the annotations on all Pods that have TLS (transport layer security) configurable components.
When you deploy Mimir or GEM, Vault Agent fetches the relevant secrets from Vault according to the annotations and mounts them to the Pod.

> **Note:** Vault and Vault Agent must to be running already.
```

Reviewer 2: Vetted this with @fayzal-g.

Example values file:

```yaml
vaultAgent:
  enabled: true
  roleName: "test-role"
  clientCertPath: "client/cert/path"
  clientKeyPath: "client/key/path"
  serverCertPath: "server/cert/path"
  serverKeyPath: "server/key/path"
  caCertPath: "ca/cert/path"
```

> Example Deployment YAML:

Reviewer 2: ```suggestion
Generated Kubernetes Deployment YAML file based on the preceding example:
```

  annotations: {}
  namespace: "default"
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: mimir
      app.kubernetes.io/instance: release-name
      app.kubernetes.io/component: component-name
  strategy:
    type: RollingUpdate
  template:
    metadata:
      annotations:
        vault.hashicorp.com/agent-inject: "true"
        vault.hashicorp.com/role: "test-role"
        vault.hashicorp.com/agent-inject-secret-client.crt: "client/cert/path"
        vault.hashicorp.com/agent-inject-secret-client.key: "client/key/path"
        vault.hashicorp.com/agent-inject-secret-server.crt: "server/cert/path"
        vault.hashicorp.com/agent-inject-secret-server.key: "server/key/path"
        vault.hashicorp.com/agent-inject-secret-root.crt: "ca/cert/path"
```

`vault.hashicorp.com/agent-inject-secret-<FILENAME>: '<PATH>'` tells the Vault Agent where to find the secret, and the name of the file to write the secret to. For example: `vault.hashicorp.com/agent-inject-secret-client.crt: 'client/cert/path'` will look for the secret at the path `client/cert/path` within Vault, and mount this secret to the pod as `client.crt` in the `/vault/secrets/` directory.

> For more information about Vault and Vault Agent, see this [blog.](https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar)

Reviewer 2: ```suggestion
For more information about Vault and Vault Agent, see [Injecting Vault Secrets Into Kubernetes Pods via a Sidecar](https://www.hashicorp.com/blog/injecting-vault-secrets-into-kubernetes-pods-via-a-sidecar).
```

> To configure TLS in Mimir, refer to [Securing Grafana Mimir communications with TLS](/docs/mimir/v2.7.x/operators-guide/secure/securing-communications-with-tls.md)

Reviewer 2: We need to check this version and whether or not we need a relref.
